library/core/src/main/java/com/google/android/exoplayer2/upstream/ContentDataSource.java

```diff
@@ -89,7 +89,7 @@ public long open(DataSpec dataSpec) throws ContentDataSourceException {
         long assetFileDescriptorLength = assetFileDescriptor.getLength();
         if (assetFileDescriptorLength == AssetFileDescriptor.UNKNOWN_LENGTH) {
           // The asset must extend to the end of the file.
-          bytesRemaining = inputStream.available();
+          bytesRemaining = ((FileInputStream) inputStream).getChannel().size();
           if (bytesRemaining == 0) {
             // FileInputStream.available() returns 0 if the remaining length cannot be determined,
```

Comment:
This comment doesn't seem to make sense given then change being made? Does the if block itself still make sense?

Reply:
bytesRemaining = inputStream.available()

- for the momen when you open Pipe this call can return 0. 

bytesRemaining = ((FileInputStream) inputStream).getChannel().size();

- FileDescription for correct file in file system, this call will return correct file size. 

Again. I do not understand, why you asking this. Propbably need more testing.

Reply:
In the previous code we actually called `inputStream.available()` and then did something conditionally on the value being 0. We're not making an `inputStream.available()` call any more, so the comment doesn't make sense. Can `size()` return 0? If not why is this if block still here? If so the comment needs changing.